File: RegistrationSystem.cpp
#include <iostream>
#include <string>
#include <unordered_map>
using namespace std;

int main() {
    int n;
    cin >> n;
    
    unordered_map<string, int> map;
    
    for (int i = 0; i < n; i++) {
        string name;
        cin >> name;
        
        if (map.find(name) != map.end()) {
            int count = map[name];
            string newName = name + to_string(count);
            while (map.find(newName) != map.end()) {
                count++;
                newName = name + to_string(count);
            }
            cout << newName << endl;
            map[name] = count + 1;
            map[newName] = 1;
        } else {
            cout << "OK" << endl;
            map[name] = 1;
        }
    }
    
    return 0;
}

File: RegistrationSystem.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class RegistrationSystem {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int n = scanner.nextInt();
        scanner.nextLine();
        
        Map<String, Integer> map = new HashMap<>();
        
        for (int i = 0; i < n; i++) {
            String name = scanner.nextLine();
            
            if (map.containsKey(name)) {
                int count = map.get(name);
                String newName = name + count;
                while (map.containsKey(newName)) {
                    count++;
                    newName = name + count;
                }
                System.out.println(newName);
                map.put(name, count + 1);
                map.put(newName, 1);
            } else {
                System.out.println("OK");
                map.put(name, 1);
            }
        }
        
        scanner.close();
    }
}